lib/vmpooler/pool_manager.rb

```diff
@@ -289,161 +289,168 @@ def destroy_vm(vm, pool)
+            (! $redis.sismember('vmpooler__ready__' + pool['name'], vm['name'])) &&
+            (! $redis.sismember('vmpooler__pending__' + pool['name'], vm['name'])) &&
+            (! $redis.sismember('vmpooler__completed__' + pool['name'], vm['name'])) &&
+            (! $redis.sismember('vmpooler__discovered__' + pool['name'], vm['name']))
 
-                $logger.log('s', "[?] [#{pool['name']}] '#{vm['name']}' added to 'discovered' queue")
-              end
+            $redis.sadd('vmpooler__discovered__' + pool['name'], vm['name'])
 
-              inventory[vm['name']] = 1
-            end
-          rescue
+            $logger.log('s', "[?] [#{pool['name']}] '#{vm['name']}' added to 'discovered' queue")
           end
 
-          # RUNNING
-          $redis.smembers('vmpooler__running__' + pool['name']).each do |vm|
-            if inventory[vm]
-              if pool['running_ttl']
-                begin
-                  check_running_vm(vm, pool['name'], pool['running_ttl'])
-                rescue
-                end
-              else
-                begin
-                  check_running_vm(vm, pool['name'], '720')
-                rescue
-                end
-              end
-            end
-          end
+          inventory[vm['name']] = 1
+        end
+      rescue
+      end
 
-          # READY
-          $redis.smembers('vmpooler__ready__' + pool['name']).each do |vm|
-            if inventory[vm]
-              begin
-                check_ready_vm(vm, pool['name'], pool['ready_ttl'] || 0)
-              rescue
-              end
+      # RUNNING
+      $redis.smembers('vmpooler__running__' + pool['name']).each do |vm|
+        if inventory[vm]
+          if pool['running_ttl']
+            begin
+              check_running_vm(vm, pool['name'], pool['running_ttl'])
+            rescue
+            end
+          else
+            begin
+              check_running_vm(vm, pool['name'], '720')
+            rescue
             end
           end
+        end
+      end
 
-          # PENDING
-          $redis.smembers('vmpooler__pending__' + pool['name']).each do |vm|
-            unless pool['timeout']
-              if $config[:config]['timeout']
-                pool['timeout'] = $config[:config]['timeout']
-              else
-                pool['timeout'] = 15
-              end
-            end
+      # READY
+      $redis.smembers('vmpooler__ready__' + pool['name']).each do |vm|
+        if inventory[vm]
+          begin
+            check_ready_vm(vm, pool['name'], pool['ready_ttl'] || 0)
+          rescue
+          end
+        end
+      end
 
-            if inventory[vm]
-              begin
-                check_pending_vm(vm, pool['name'], pool['timeout'])
-              rescue
-              end
-            end
+      # PENDING
+      $redis.smembers('vmpooler__pending__' + pool['name']).each do |vm|
+        unless pool['timeout']
```

Comment:
`pool['timeout']` is not used unless the VM is present in `inventory`. This should be moved within the `if inventory[vm]` block.